web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java

```diff
@@ -0,0 +1,203 @@
+        String password = CoreResources.getField("insight.account.password");
+
+        // get Session ID
+        HashMap<String, String> headers = new HashMap<String, String>();
+        headers.put("Content-Type", "application/json; charset=UTF-8");
+        String jsonCredential = "{\"username\":\"" + username + "\",\"password\":\"" + password + "\"}";
+        String strSession = doPost(insightURL + "/api/session", "POST", headers,
+                jsonCredential);
+        JSONObject session = new JSONObject(strSession);
+
+        // get content from insight
+        headers = new HashMap<String, String>();
+        headers.put("X-Metabase-Session", session.get("id").toString());
+
+        String parameter = "parameters=[{\"type\":\"category\"," +
+                "\"target\":[\"dimension\",[\"template-tag\",\"original_id\"]]," +
+                "\"value\":[\"" + participantLabels[0] + "\"]}," +
+                "{\"type\":\"category\"," +
+                "\"target\":[\"dimension\",[\"template-tag\",\"replicate_id\"]]," +
+                "\"value\":[\"" + participantLabels[1] + "\"]}]";
+        String response = doPost(insightURL + "/api/card/" + reportID + "/query/csv?" + parameter, "POST", headers,
+                null);
+
+        Study site = studyDao.findByOcOID(studyOid);
+        if (site != null) {
+            Study parentStd = site.getStudy();
+            if (parentStd != null) {
+                utilService.setSchemaFromStudyOid(studyOid);
+                UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
+                UserAccount userAccount = userAccountDao.findById(userAccountBean.getId());
+
+                String fileName = saveToFile(response.toString(), participantLabel);
+                if (fileName != null) {
+                    JobDetail jobDetail= userService.persistJobCreated(parentStd, site, userAccount, JobType.INSIGHT_REPORT, fileName);
+                    userService.persistJobCompleted(jobDetail, fileName);
+                }
+            }
+        }
+
+        logger.info("REST request to POST Insight Report");
+        return new ResponseEntity(HttpStatus.OK);
+    }
+
+    public String[] getLabels(String label) {
+        // original label/id not contain replicaSubString
+        // replica label/id contain replicaSubString
+        String[] labels = new String[2];
+        if (label.indexOf(replicaSubstring) > -1) {
+            labels[0] = label.split(replicaSubstring)[0];
+            labels[1] = label;
+        } else {
+            labels[0] = label;
+            labels[1] = label + replicaSubstring;
+        }
+        return labels;
+    }
+
+    public String doPost(String api, String method, HashMap<String, String> headers,
+                         String body) throws Exception {
+        URL url = new URL(api);
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
Can you please use RestTemplate? See examples in other places in the project.

Reply:
code updated